### PR TITLE
chore(KFLUXUI-717): enhanced commits search

### PR DESCRIPTION
**Pull request by janaki29**
[KFLUXUI-717](https://redhat.atlassian.net/browse/KFLUXUI-717)

## Description

Earlier, the search on the Commits tab/page only filtered the pipeline runs already loaded on the screen. This prevented users from finding older commits.

This PR is to change that behaviour. The search bar now queries the backend directly to find matching commits across all pipeline runs, including those in the cluster, tekton-results, and kubearchive.

Rather than embedding the search filter into the usePipelineRunsV2 hook's selector (which caused two problems: disrupting the cluster watch on every keystroke, and being entirely skipped when the KubeArchive feature flag is enabled), the solution adds a separate, dedicated Tekton Results search query in CommitsListView. This query:

- Runs only when the user types a search term
- Queries Tekton Results directly, bypassing both usePipelineRunsV2 and the KubeArchive feature flag
- Merges matching historical results into the existing pipeline run data
- Shows a loading state while the search query is in flight

## Type of change

- [x] Feature
- [ ] Bugfix
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## Browser conformance: 

- [ ] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

[KFLUXUI-717]: https://redhat.atlassian.net/browse/KFLUXUI-717?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

## Summary by CodeRabbit

* **New Features**
  * Commit/name search now queries external results, supports case-insensitive matching, and recognizes PR numbers with/without “#”. Added an option to pass search terms through run-list queries.

* **User Experience**
  * Removed duplicate client-side name filtering; status filters and pagination preserved.
  * "Load more" spinner appears only when results exist.
  * Toolbar renders in no-data view; "Add integration test" action distinguishes empty vs toolbar contexts.

* **Tests**
  * Added/updated tests covering commit-search behavior, external query integration, and search-reset flows.

### CODE REVIEW

**Review by rrosatti — COMMENTED** [review]
I left some minor comments. Overall code looks good. Also tested using both tekton-results and kubearchive, and they seem to be working as expected!

File: src/utils/pipeline-run-commit-search.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import { PipelineRunLabel } from '../consts/pipelinerun';
+import { PipelineRunKind } from '../types';
+
+/**
+ * In-memory filter aligned with `commitSearchFilter` in `tekton-results.ts` for cluster and
+ * KubeArchive pipeline run lists (Tekton Results uses the CEL `commitSearchFilter` instead).
+ */
+export const pipelineRunMatchesCommitSearch = (
```
Comment (rrosatti) [inline src/utils/pipeline-run-commit-search.ts]: how about moving this function to the `src/utils/pipelinerun-utils.ts` file? :)

File: src/components/Commits/CommitsListPage/CommitsListView.tsx [hunk 2]
```diff
@@ -111,10 +111,14 @@ const CommitsListView: React.FC<React.PropsWithChildren<CommitsListViewProps>> =
             matchLabels: {
               [PipelineRunLabel.APPLICATION]: applicationName,
               ...(componentName ? { [PipelineRunLabel.COMPONENT]: componentName } : {}),
+              ...(nameFilter.trim()
```
Comment (rrosatti) [inline src/components/Commits/CommitsListPage/CommitsListView.tsx]: is this change really needed? it might cause unexpected results for the user, because the "non-build pipeline runs" would suddenly disappear when `nameFilter` is updated 🤔

Comment (janaki29) [reply src/components/Commits/CommitsListPage/CommitsListView.tsx]: Removed. It was a scoping mistake during the refactor.

**Review by JoaoPedroPP — COMMENTED** [review]
When I try with an invalid input, for example: `#abcd1234` the page did not find any result, which is expected but the search bar faded and I was unable to try new input unless I refresh the page.

Image

**Comment by janaki29** [comment]
@JoaoPedroPP toolbar added for empty data result.

**Review by rrosatti — COMMENTED** [review]
overall code looks good, but could you please take a look at the PR checks?

1. some unit tests are failing
2. Codecov is saying only ~36% of this patch is covered by tests. Could you please check if it's possible to increase the coverage? 

thx!

**Review by JoaoPedroPP — DISMISSED** [review]
LGTM

UI is working as expected. Nice jobs

**Review by rrosatti — COMMENTED** [review]
the search input (whole filter toolbar) is gone if it doesn't return any results but it's still fetching: 

https://github.com/user-attachments/assets/e8069275-5631-4ee4-a2d9-2c1ae6981be3

how about keeping the filter search there while data is being fetched and there's still no results? :)

File: src/utils/pipeline-run-commit-search.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  if (!term) return true;
+
+  const labels = plr.metadata?.labels ?? {};
+  const annotations = plr.metadata?.annotations ?? {};
+  const prSubterm = term.replace('#', '');
+
+  const fieldContainsTerm = (val: unknown) =>
+    typeof val === 'string' && val.toLowerCase().includes(term);
+
+  return (
```
Comment (StanislavJochman) [inline src/utils/pipeline-run-commit-search.ts:22]: why here you use || and down bellow you use OR?

File: src/components/Commits/CommitsListPage/__tests__/CommitsListView.spec.tsx [hunk 4]
```diff
@@ -249,31 +302,71 @@ describe('CommitsListView', () => {
...
     renderWithQueryClient(
       <FilterContextProvider filterParams={['name', 'status']}>
         <CommitsListView applicationName="purple-mermaid-app" componentName="sample-component" />
       </FilterContextProvider>,
     );
 
     expect(usePipelineRunsV2Mock).toHaveBeenCalledWith(
-      'test-ns', // namespace
-      {
+      'test-ns',
```
Comment (StanislavJochman) [inline src/components/Commits/CommitsListPage/__tests__/CommitsListView.spec.tsx]: keep the comment.

**Comment by StanislavJochman** [comment]
@coderabbitai resume

**Review by StanislavJochman — DISMISSED** [review]
LGTM. Please address comments + Rodrigo suggestion.

File: src/utils/pipeline-run-commit-search.ts (hunk 3 above)
Comment (janaki29) [reply src/utils/pipeline-run-commit-search.ts:22]: the two files are operating in different languages/contexts, making || and OR() the correct choice in each respective location.

**Review by rrosatti — APPROVED** [review]
LGTM! nice job :)

**Review by JoaoPedroPP — APPROVED** [review]
LGTM

UI is working as expected.
